android/java/org/chromium/chrome/browser/preferences/BraveSearchEngineUtils.java

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.preferences;
+
+import android.content.SharedPreferences;
+
+import org.chromium.base.ContextUtils;
+import org.chromium.base.Log;
+import org.chromium.chrome.browser.search_engines.TemplateUrlServiceFactory;
+import org.chromium.chrome.browser.tabmodel.TabModelSelector;
+import org.chromium.components.search_engines.TemplateUrl;
+
+public class BraveSearchEngineUtils {
+    private static final String PRIVATE_DSE_KEYWORD = "private_dse_keyword";
+    private static final String STANDARD_DSE_KEYWORD = "standard_dse_keyword";
+    private static final String PRIVATE_DSE_SHORTNAME = "private_dse_shortname";
+    private static final String STANDARD_DSE_SHORTNAME = "standard_dse_shortname";
+
+    static public String getDSEShortName(boolean isPrivate) {
+        String defaultSearchEngineName = null;
+        TemplateUrl dseTemplateUrl =
+                TemplateUrlServiceFactory.get().getDefaultSearchEngineTemplateUrl();
+        if (dseTemplateUrl != null)
+            defaultSearchEngineName = dseTemplateUrl.getShortName();
+
+        return ContextUtils.getAppSharedPreferences().getString(
+            isPrivate ? PRIVATE_DSE_SHORTNAME : STANDARD_DSE_SHORTNAME, defaultSearchEngineName);
+    }
+
+    static public String getDSEKeyword(boolean isPrivate) {
+        String defaultSearchEngineName = null;
+        TemplateUrl dseTemplateUrl =
+                TemplateUrlServiceFactory.get().getDefaultSearchEngineTemplateUrl();
+        if (dseTemplateUrl != null)
+            defaultSearchEngineName = dseTemplateUrl.getKeyword();
+
+        return ContextUtils.getAppSharedPreferences().getString(
+            isPrivate ? PRIVATE_DSE_KEYWORD : STANDARD_DSE_KEYWORD, defaultSearchEngineName);
+    }
+
+    static public void updateActiveDSEWith(boolean isPrivate) {
```

Comment:
I think the name here is confusing or maybe I just don't understand it. Maybe it just doesn't need `With` in there? Probably goes for all the occurrences of this.

Reply:
Yes, it would be better w/o `With`. Removed.